tests/test_model.py

```diff
@@ -176,3 +177,17 @@ def test_models_batch_settings_are_ignored_in_ensemble(
     step_cmd = step.step_cmds[0]
     assert any("srun" in tok for tok in step_cmd)  # call the model using run settings
     assert not any("sbatch" in tok for tok in step_cmd)  # no sbatch in sbatch
+
```

Comment:
Split this test up (right now you're testing both the happy and bad routes) and parametrize the bad routes.

```
import numpy as np

@pytest.mark.parametrize("dtype", [int, np.float32, str]):
def test_good_params(dtype):
  params = {"foo": dtype(0.6)}
  assert all(isinstance(val, str) for val in _read_model_parameters(params).values())
  
@pytest.mark.parametrize("bad_val", [["eggs"], {"n": 5}, object)
def test_bad_model_params(bad_val):
  with pytest.raises(TypeError):
    _read_model_parameters({"foo": bad_val})
```

Reply:
changed